### PR TITLE
Changes to LLR output and handling of alt_fit

### PR DESCRIPTION
**Pull request by steven-j-wren**
I have changed the output so that the trials are not put in to a dictionary with 'trials' as the key. This is so that the output 'false_h_settings' and 'llh_null' that also go in to this dictionary are not overwritten when the trials are saved, as they were previously.

Also, I added a deepcopy for when fix_non_atm_params is used, since it was causing the non_atm_params to be fixed in template_settings['params'], so when one called get_llh_hypothesis for the alt_fit it was only minimising over dm31 and th23.

### CODE REVIEW

File: pisa/analysis/llr/LLROptimizerAnalysis.py [hunk 1]
```diff
@@ -228,7 +228,8 @@ def get_llh_hypothesis(
         logging.info("Running false hierarchy best fit...")
         output[data_tag]["false_h_best_fit"] = {}
 
-        false_h_params = fix_non_atm_params(template_settings['params'])
+        proxy_settings = deepcopy(template_settings['params'])
```
Comment (thehrh) [inline pisa/analysis/llr/LLROptimizerAnalysis.py]: This isn't necessary anymore: see commit eff2821f856b4cfac5ad138f3cb208f11619533c.

**Comment by steven-j-wren** [comment]
Sorry to do a double commit there. I had commented some lines for debugging that I forgot to remove.

File: pisa/analysis/llr/LLROptimizerAnalysis.py [hunk 2]
```diff
@@ -228,6 +228,7 @@ def get_llh_hypothesis(
         logging.info("Running false hierarchy best fit...")
         output[data_tag]["false_h_best_fit"] = {}
 
+        #proxy_settings = deepcopy(template_settings['params'])
```
Comment (sboeser) [inline pisa/analysis/llr/LLROptimizerAnalysis.py]: If this line is not needed, please remove it.

**Comment by sboeser** [comment]
Hi Ste, not quite sure what the reason is for storing the trials under the additional keyword `trials` when this is the only thing we actually store in the outer `true_h_fiducial` or `false_h_best_fit` dictionary?

**Comment by steven-j-wren** [comment]
It's the only output under true_h_fiducial, but it's not the only output under false_h_best_fit. There is also:

output[data_tag]['false_h_best_fit']['false_h_settings'] = false_h_settings
output[data_tag]['false_h_best_fit']['llh_null'] = llh_data

These aren't shown in the diff since they were already in the file. It's just they ended up being overwritten by the way the trials were being output. i.e.

output[data_tag]['false_h_best_fit'] = trials

overwrote them, so I just added another key.

**Comment by sboeser** [comment]
Hi Ste, thanks for clarifying and fixing this bug!

**Comment by sboeser** [comment]
Oh - I forgot: does this require updates in the plotting functions for LLR by Tim? Maybe @jllanfranchi knows?

**Comment by steven-j-wren** [comment]
It probably will do. If the plotting functions have already been changed to accommodate the alteration in the treatment of the alternative hierarchy hypothesis, then it should be fairly trivial to just tell it to look one key further in the output files. But, I haven't used the LLR plotting scripts in a long time.

**Pull request merged**